tools/ci_build/github/azure-pipelines/win-gpu-ci-pipeline.yml

```diff
@@ -88,12 +88,13 @@ stages:
         BuildConfig: 'RelWithDebInfo'
         EnvSetupScript: setup_env_cuda_11.bat
         buildArch: x64
-        additionalBuildFlags: --gen_doc validate --skip_tests --enable_pybind --use_dml --use_cuda --cuda_version=11.6 --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6" --enable_cuda_profiling --cmake_extra_defines CMAKE_CUDA_ARCHITECTURES=52 --cmake_extra_defines onnxruntime_BUILD_UNIT_TESTS=OFF
+        # note: need to specify `--gen_doc` when creating the build config so it has to be in additionalBuildFlags
+        additionalBuildFlags: --gen_doc --skip_tests --enable_pybind --use_dml --use_cuda --cuda_version=11.6 --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6" --cmake_extra_defines CMAKE_CUDA_ARCHITECTURES=52 --cmake_extra_defines onnxruntime_BUILD_UNIT_TESTS=OFF
         msbuildPlatform: x64
         isX86: false
         job_name_suffix: x64_RelWithDebInfo
         RunOnnxRuntimeTests: false
         RunStaticCodeAnalysis: false
+        GenerateDocumentation: true
         ORT_EP_NAME: CUDA # It doesn't really matter which EP is selected here since this stage is for documentation.
-        MachinePool: Win-CPU-2019
-        DocUpdateNeeded: true
+        MachinePool: onnxruntime-Win2019-GPU-dml
```

Comment:
Now we can run the build on a CPU pool.

Reply:
CUDA EP seems to have the same issue.

RuntimeError: D:\a\_work\1\s\onnxruntime\core\providers\cuda\cuda_call.cc:124 onnxruntime::CudaCall D:\a\_work\1\s\onnxruntime\core\providers\cuda\cuda_call.cc:117 onnxruntime::CudaCall CUDA failure 100: no CUDA-capable device is detected ; GPU=48 ; hostname=5f152811c000006 ; expr=cudaSetDevice(info_.device_id); 

https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=795876&view=logs&j=7f366e99-16b2-52cc-e1ff-653af284e397&t=6deb7cde-8f5a-55b8-05df-c4291cdac549

Reply:
> CUDA EP seems to have the same issue.

Doh, I anticipate this will be an issue with other future EP's too then if we want to merge their documentation too into OperatorKernels.md. So having a means to just enumerate an EP's kernels without fully initializing an EP's device would be useful. 🤔